client/v2/client.go

```diff
@@ -413,16 +414,45 @@ func (uc *udpclient) Write(bp BatchPoints) error {
 	for _, p := range bp.Points() {
 		pointstring := p.pt.RoundedString(d) + "\n"
 
-		// Write and reset the buffer if we reach the max size
 		if b.Len()+len(pointstring) >= uc.payloadSize {
 			if _, err := uc.conn.Write(b.Bytes()); err != nil {
 				return err
 			}
 			b.Reset()
 		}
 
-		if _, err := b.WriteString(pointstring); err != nil {
-			return err
+		if b.Len()+len(pointstring) <= uc.payloadSize {
+			if _, err := b.WriteString(pointstring); err != nil {
+				return err
+			}
+		} else {
+			if p.Time().IsZero() {
+				if len(pointstring) > MaxPointSize {
+					return fmt.Errorf("point size is greater than %dB", MaxPointSize)
+				}
+				if _, err := b.WriteString(pointstring); err != nil {
+					return err
+				}
+			} else {
```

Comment:
Same here... `continue` and removing the `else` would remove a bunch of indentation.